test/routing/route_test.cc

```diff
@@ -726,6 +729,582 @@ TEST_F(RouteLaneSRelationTest, EvaluateComputeLaneSRangeRelationSameLaneSRanges)
+    EXPECT_CALL(lane_b_, DoGetOrientation(_)).WillRepeatedly(Return(kRotation));
+    // Segment one.
+    SetUpLane(&lane_c_, kLaneIdC, nullptr, nullptr, &segment_one_, kLaneBounds, kLaneLength);
+    EXPECT_CALL(lane_c_, DoToInertialPosition(Matches(kStartLanePositionPhase)))
+        .WillRepeatedly(Return(api::InertialPosition{kLaneLength, 0., 0.}));
+    EXPECT_CALL(lane_c_, DoGetOrientation(_)).WillRepeatedly(Return(kRotation));
+
+    EXPECT_CALL(id_index_, DoGetLane(kLaneIdA)).WillRepeatedly(Return(&lane_a_));
+    EXPECT_CALL(id_index_, DoGetLane(kLaneIdB)).WillRepeatedly(Return(&lane_b_));
+    EXPECT_CALL(id_index_, DoGetLane(kLaneIdC)).WillRepeatedly(Return(&lane_c_));
+    EXPECT_CALL(*(road_geometry_ptr_), DoById()).WillRepeatedly(ReturnRef(id_index_));
+    const std::vector<api::RoadPosition> kStartRoadPositionsPhaseZero{
+        api::RoadPosition{&lane_a_, kStartLanePositionPhase}};
+    const std::vector<api::RoadPosition> kEndRoadPositionsPhaseZero{api::RoadPosition{&lane_b_, kEndLanePositionPhase}};
+    const std::vector<api::RoadPosition> kStartRoadPositionsPhaseOne{
+        api::RoadPosition{&lane_c_, kStartLanePositionPhase}};
+    const std::vector<api::RoadPosition> kEndRoadPositionsPhaseOne{api::RoadPosition{&lane_c_, kEndLanePositionPhase}};
+
+    phase_zero_ = std::make_unique<Phase>(kPhaseZeroIndex, kLaneSRangeTolerance, kStartRoadPositionsPhaseZero,
+                                          kEndRoadPositionsPhaseZero, kLaneSRangesPhaseZero, road_network_.get());
+    phase_one_ = std::make_unique<Phase>(kPhaseOneIndex, kLaneSRangeTolerance, kStartRoadPositionsPhaseOne,
+                                         kEndRoadPositionsPhaseOne, kLaneSRangesPhaseOne, road_network_.get());
+    EXPECT_CALL(*road_geometry_ptr_, do_linear_tolerance()).WillRepeatedly(Return(kLinearTolerance));
+  }
+};
+
+TEST_F(RouteComputeLaneSRouteCaseCTest, ComputeLaneSRoute) {
+  const api::RoadPosition kRoadPositionA{&lane_a_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionB{&lane_b_, kQueryLanePosition};
+  const api::RoadPosition kRoadPositionC{&lane_c_, kQueryLanePosition};
+  const api::LaneSRoute kExpectedLaneSRouteFromA({kLaneSRangeA, kLaneSRangeB, kLaneSRangeC});
+  const api::LaneSRoute kExpectedLaneSRouteFromB({kLaneSRangeB, kLaneSRangeC});
+  const api::LaneSRoute kExpectedLaneSRouteFromC({kLaneSRangeC});
+  const Route dut({*phase_zero_, *phase_one_}, road_network_.get());
+
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromA, dut.ComputeLaneSRoute(kRoadPositionA))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromB, dut.ComputeLaneSRoute(kRoadPositionB))));
+  EXPECT_TRUE(AssertCompare(IsEqual(kExpectedLaneSRouteFromC, dut.ComputeLaneSRoute(kRoadPositionC))));
+}
+
+// Case D:
+// - Feature under evaluation: can move to the right at the end in the same Phase.
+// - Topology:
+// <pre>
+//
+// x----A>----x----B>----*
+//            *----C>----x
```

Comment:
Can we add a test case involving multiple moves to the right?

```
// x----A>----x----B>----*
//            *----C>----*
//            *----D>----x
```

Ditto for multiple moves to the left.

Reply:
Added in c7b8e5b